testsuite/regressiontests/issue-1703.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+#
+# issue-1703.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+This script ensures that importing NEST after mpi4py does not lead
+to a segmentation fault when simulating.
+
+This is a regression test for GitHub issue 1703.
+"""
+
+import sys
+import subprocess as sp
+import shlex
+
+EXIT_SUCCESS = 0
+EXIT_SKIPPED = 200
+EXIT_FAILURE = 127
+EXIT_SEGFAULT = 139
+
+try:
+    import nest
+    from mpi4py import MPI
+except ImportError:
+    # Skip test if mpi4py is not installed, or if NEST is
+    # installed without Python support.
```

Comment:
I would move this comment before the try and explain the logic more clearly: here, you test that nest and mpi4py are available (and loading mpi4py after nest is not the problem we are regression-testing here), below comes the actual test. I got at first confused by the try block.